### PR TITLE
=doc #16989 update http bind documentation

### LINKED ISSUE #16989: Http().bind documentation
A couple of suggestions:
* Update [HttpServerExampleSpec](https://github.com/akka/akka/blob/e30c3e6492fdffb20ee8baa488611213b5a66755/akka-docs-dev/rst/scala/code/docs/http/HttpServerExampleSpec.scala) so it shows how to get the `ServerBinding` object. It took me a while to work this out, because I'm a noob at Akka Streams. :) Since 1.0-M4 I see the docs are a bit clearer because there's now a `bindingFuture` variable. To make it even easier for noobs like me, it may help to add a type annotation to this variable, i.e. `Future[ServerBinding]`.
* Either remove [TestServer](https://github.com/akka/akka/blob/e30c3e6492fdffb20ee8baa488611213b5a66755/akka-http-tests/src/test/scala/akka/http/server/TestServer.scala) or update it to follow current best practice for binding. In 1.0-M4, TestServer doesn't work because of #16972 and it took me a while to find the docs (and `HttpServerExampleSpec`) with the correct information.

### PR DESCRIPTION
**Pull request by 2m**
Fixes #16989.

### CODE REVIEW

**Comment by patriknw** [comment]
LGTM

**Comment by rkuhn** [comment]
LGTM

File: akka-docs-dev/rst/scala/code/docs/http/HttpServerExampleSpec.scala [hunk 1]
```diff
@@ -21,11 +23,13 @@ class HttpServerExampleSpec
...
-    val serverSource = Http(system).bind(interface = "localhost", port = 8080)
-    serverSource.runForeach { connection => // foreach materializes the source
+    val serverSource: Source[Http.IncomingConnection, Future[Http.ServerBinding]] =
+      Http(system).bind(interface = "localhost", port = 8080)
+    val bindingFuture: Future[Http.ServerBinding] = serverSource.to(Sink.foreach { connection =>
+      // foreach materializes the source
       println("Accepted new connection from " + connection.remoteAddress)
       // ... and then actually handle the connection
-    }
+    }).run()
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/http/HttpServerExampleSpec.scala]: runForeach? Or we wanted to show the full version here?

**Comment by drewhk** [comment]
LGTM. Is there a Java counterpart?

**Pull request merged**